internal/services/appservice/windows_function_app_slot_resource.go

```diff
@@ -282,13 +266,28 @@ func (r WindowsFunctionAppSlotResource) Create() sdk.ResourceFunc {
 			}
 
 			client := metadata.Client.AppService.WebAppsClient
+			functionAppId, err := parse.FunctionAppID(functionAppSlot.FunctionAppID)
+			if err != nil {
+				return err
+			}
+
 			aseClient := metadata.Client.AppService.AppServiceEnvironmentClient
 			servicePlanClient := metadata.Client.AppService.ServicePlanClient
 			subscriptionId := metadata.Client.Account.SubscriptionId
 
-			id := parse.NewFunctionAppSlotID(subscriptionId, functionAppSlot.ResourceGroup, functionAppSlot.FunctionAppName, functionAppSlot.Name)
-
-			servicePlanId, err := parse.ServicePlanID(functionAppSlot.ServicePlanId)
+			id := parse.NewFunctionAppSlotID(subscriptionId, functionAppId.ResourceGroup, functionAppId.SiteName, functionAppSlot.Name)
+			functionApp, err := client.Get(ctx, functionAppId.ResourceGroup, functionAppId.SiteName)
+			if err != nil {
+				return fmt.Errorf("reading parent Windows Function App for %s: %+v", id, err)
+			}
+			if functionApp.Location == nil {
+				return fmt.Errorf("could not determine location for %s: %+v", id, err)
+			}
+			props := functionApp.SiteProperties
+			if props == nil || props.ServerFarmID == nil {
+				return fmt.Errorf("could not determine Service Plan ID for %s: %+v", id, err)
+			}
+			servicePlanId, err := parse.ServicePlanID(*props.ServerFarmID)
```

Comment:
does this want to be parsed insensitively?

Reply:
As discussed offline, this will be addressed later